teos/src/dbm.rs

```diff
@@ -1175,7 +1193,61 @@ mod tests {
         dbm.store_tracker(uuid, &tracker).unwrap();
 
         // We should get all the appointments matching our locator back except from the triggered one
-        assert_eq!(dbm.load_appointments(Some(locator)), appointments);
+        assert_eq!(dbm.load_appointments(Some((locator, None))), appointments);
+    }
+
+    #[test]
+    fn test_load_appointments_with_locator_and_user_id() {
+        let dbm = DBM::in_memory().unwrap();
+
+        let mut appointments = HashMap::new();
+        let dispute_tx = get_random_tx();
+        let dispute_txid = dispute_tx.txid();
+        let locator = Locator::new(dispute_txid);
+
+        // Create user id
+        let user_id = get_random_user_id();
+        let user = UserInfo::new(AVAILABLE_SLOTS, SUBSCRIPTION_START, SUBSCRIPTION_EXPIRY);
+        dbm.store_user(user_id, &user).unwrap();
+
+        // Create and store a particular appointment
+        let (uuid, appointment) =
+            generate_dummy_appointment_with_user(user_id, Some(&dispute_txid));
+        dbm.store_appointment(uuid, &appointment).unwrap();
+        appointments.insert(uuid, appointment.clone());
+
+        // Create random appointments
+        for _ in 1..11 {
+            let (uuid, appointment) = generate_dummy_appointment_with_user(user_id, None);
+            dbm.store_appointment(uuid, &appointment).unwrap();
+            appointments.insert(uuid, appointment);
+        }
+
+        // Verify that no appointment is returned if there is not an exact match of user_id + locator
+        assert_eq!(
+            dbm.load_appointments(Some((locator, Some(get_random_user_id()))),),
+            HashMap::new()
+        );
+        assert_eq!(
+            dbm.load_appointments(Some((get_random_locator(), Some(user_id))),),
+            HashMap::new()
+        );
+
+        // Verify that the expected appointment is returned, if the correct user_id and locator is given
+        assert_eq!(
+            dbm.load_appointments(Some((locator, Some(user_id))),),
+            HashMap::from([(uuid, appointment)])
+        );
+
+        // Create a tracker from the existing appointment
+        let tracker = get_random_tracker(user_id, ConfirmationStatus::InMempoolSince(100));
+        dbm.store_tracker(uuid, &tracker).unwrap();
+
+        // Verify that an appointment is not returned, if it is triggered (there's a tracker for it)
+        assert_eq!(
+            dbm.load_appointments(Some((locator, Some(user_id))),),
+            HashMap::new()
+        );
```

Comment:
Ummmm, I wonder whether this part of the test succeeded? Because you created a random tracker here, not related to the appointment with the dispute tx above.

Reply:
Aha, so this works because we stored it in the dbm with the same `uuid`.
I think it would be better to construct the tracker like this:
```rust
        let tracker = TransactionTracker::new(
            Breach::new(dispute_tx.clone(), get_random_tx()),
            user_id,
            ConfirmationStatus::ConfirmedIn(100),
        );
```
to make such a relationship between the appointment and its tracker clearer.